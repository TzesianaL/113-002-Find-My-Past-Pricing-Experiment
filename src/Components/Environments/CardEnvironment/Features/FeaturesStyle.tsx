import styled from 'styled-components'

/**
 * Styled component is used to decorate the features component.
 * Props are passed from ThemeProvider component.
 */

export const FeaturesStyle = styled.ul`
  color: black;
  font-size: ${(props) => props.theme.fontSize}em;
  font-family: ${(props) => props.theme.fontFamily};

  padding-left: 2em;
  padding-right: 2em;
  /* padding: 0; */
  width: calc(100% - 4em);
  li {
    padding-left: 0.25em;
    // margin-bottom: 1.5em;

    ::marker {
      content: '${(props) => props.theme.featureBullet || 'âœ“'}';
    }
  }
`

export const DropdownHeader = styled.div`
  background-color: #f5f0eb;
  padding: 12px 16px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  width: 100%;
  box-sizing: border-box;
  font-family: ${(props) => props.theme.fontFamily};
  font-size: ${(props) => props.theme.fontSize}em;
  color: black;
  user-select: none;

  span:first-child {
    flex-grow: 1;
    text-align: center;
  }

  span:last-child {
    position: absolute;
    right: 16px;
  }

  &:hover {
    background-color: #f0ebe6;
  }
`

export const DropdownContent = styled.div<{ isExpanded: boolean }>`
  max-height: ${(props) => (props.isExpanded ? 'auto' : '0')};
  overflow: hidden;
  transition: max-height 0.3s ease;
  background-color: #f5f0eb;
`
