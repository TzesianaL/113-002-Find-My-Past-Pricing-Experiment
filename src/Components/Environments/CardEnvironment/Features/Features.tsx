import React, { FC } from 'react'
import { CSSProperties } from 'styled-components'
import { FeaturesStyle } from './FeaturesStyle'

export interface FeaturesProps {
  /**
   * Features is an array which contains all the features of string type
   */
  features: string[]
  style?: CSSProperties
}

/**
 * Component that displays features using un-order list and styled component.
 */
const Features: FC<FeaturesProps> = ({ style, features }) => {
  if (features.length === 0) {
    return null
  }

  return (
    <FeaturesStyle style={style}>
      {/* {features[0]} */}
      {/* {features && features.map((feature) => <li key={feature}>{feature}</li>)} */}
      {features &&
        features.map((feature) => (
          <div
            style={{ marginBottom: '11px' }}
            dangerouslySetInnerHTML={{ __html: feature }}
            key={feature}
          />
        ))}
    </FeaturesStyle>
  )
}

export default Features
