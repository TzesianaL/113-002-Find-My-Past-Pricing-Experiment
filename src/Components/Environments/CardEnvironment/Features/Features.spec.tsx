import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import Features from './Features'

describe('<Features />', () => {
  const featArray = ['feat1', 'feat2', 'feat3']
  it('should render correctly', () => {
    const tree = renderer.create(<Features features={featArray} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should display the text of the provided features', () => {
    const list = render(<Features features={featArray} />)
    expect(list.queryByText(/feat1/ && / feat2/ && /feat3/)).toBeInTheDocument()
  })
})
