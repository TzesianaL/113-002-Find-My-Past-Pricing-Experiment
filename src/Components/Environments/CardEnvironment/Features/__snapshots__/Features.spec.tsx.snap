// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Features /> should render correctly 1`] = `
.c0 {
  color: black;
  font-size: em;
  padding-left: 2em;
  padding-right: 2em;
  width: calc(100% - 4em);
}

.c0 li {
  padding-left: 0.25em;
  margin-bottom: 1.5em;
}

.c0 li::marker {
  content: 'âœ“';
}

<ul
  className="c0"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "feat1",
      }
    }
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "feat2",
      }
    }
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "feat3",
      }
    }
  />
</ul>
`;
